### ISSUE TITLE
REQUIRE_THROWS_MESSAGE not checking message correctly

### ISSUE THREAD

**Issue by martinfinke** [onqtam/doctest#150]
## Description
`REQUIRE_THROWS_MESSAGE` only checks if an exception was thrown, and doesn't check if the exception had the expected message.

[EDIT] I understand that the macro doesn't do what I thought. There doesn't seem to be documentation on it, and the name is IMO misleading. I would except that the message has to match the exception's `what()`. Is there a macro like this in doctest?

### Steps to reproduce
```cpp
TEST_CASE("REQUIRE_THROWS_MESSAGE")
{
    SUBCASE("Checks whether the message is correct")
    {
        auto throwException = [] {
            throw std::runtime_error("Some Message");
        };

        // This passes
        REQUIRE_THROWS_MESSAGE(throwException(), "Something else");
    }
}
```

### Extra information

* doctest version: **2.0.0**
* Operating System: **macOS 10.13.6**
* Compiler+version: **Xcode 9.4.1**

**Comment by onqtam** [comment]
Hi!

Perhaps I should improve the documentation...

As for checking the message of an exception - that is in the roadmap - there is even an open issue for it.

This might get into version 2.1 but that wouldn’t be anytime soon.

**Comment by martinfinke** [comment]
Ok, got it! Thanks for clearing that up.

**Comment by onqtam** [comment]
In the recently released version 2.1 (latest is 2.2) I added the ```CHECK_THROWS_WITH(expr, exception_string)``` macro - [changelog](https://github.com/onqtam/doctest/blob/master/CHANGELOG.md#210-2018-11-30)
